lightpack.py

```diff
@@ -173,13 +184,123 @@ def _sendAndExpectSuccess(self, command):
+		return self.getColoursFromAll()[self._ledIndex(led)]
+	getColor = getColour
+
+	def getColours(self, *args):
+		"""
+		Get the individual colours of multiple LEDs.
+
+		:returns: Dictionary of tuples of red, green, blue values (0 to 
+		255), using LED numbers as integer keys
+		"""
+		defs = [self._ledIndex(arg) for arg in args]
+		colours = self.getColoursFromAll()
+		return dict([(k, colours[k]) for k in colours if k in defs])
+	getColors = getColours
+
+	def _ledColourRead(self, snippet):
+		"""
+		Read a LED colour state snippet into RGB Tuple.
+
+		:param snippet: LED colour state snipped
+		:type snippet: str
+		:returns: Tuple with LED and tuple of red, green, blue values (0 to 
+		255)
+		"""
+		try:
+			parts = snippet.split('-', 1)
+			colours = parts[1].split(',', 2)
+			rgb = [int(x) for x in colours if x.strip()]
+			led = parts[0].strip()
+			if not led or len(rgb) != 3:
+				return None
+			return int(led), tuple(rgb)
+		except IndexError:
+			return None
+		except ValueError:
+			return None
+
+	def getColoursFromAll(self):
+		"""
+		Get the colours for all LEDs.
+
+		:returns: Dictionary of tuples of red, green, blue values (0 to 
+		255), using LED numbers as integer keys
+		"""
+		commands = self._sendAndReceivePayload('getcolors').rstrip(';')\
+			.split(';')
+		colours = {}
+		for command in commands:
+			data = self._ledColourRead(command)
+			if data is not None:
+				colours[data[0]] = data[1]
+		return colours
+	getColorsFromAll = getColoursFromAll
+
+	def getDevice(self):
+		"""
+		Get the current Lightpack device type.
+
+		:returns: string
+		"""
+		return self._sendAndReceivePayload('getdevice')
+
+	def getDevices(self, fresh=True):
+		"""
+		Get a list of compatible Lightpack device types.
+
+		If the parameter fresh (default True) is set to False, a previously 
+		cached value will be used if available.
+
+		:param fresh: fetch a new value or use cached response
+		:type fresh: boolean
+		:returns: list of strings
+		"""
+		if fresh or self._devices == []:
+			self._devices = self._sendAndReceivePayload('getdevices').rstrip(
+				';').split(';')
+		return self._devices
+
+	def getFps(self):
+		"""
+		Get the current number of frames per second.
+
+		:returns: integer
+		"""
+		return int(self._sendAndReceivePayload('getfps'))
+
+	def getMode(self):
+		"""
+		Get the mode of the current profile.
+
+		:returns: string
+		"""
+		return self._sendAndReceivePayload('getmode')
+
+	def getProfiles(self, fresh=True):
 		"""
 		Get a list of profile names.
 
+		If the parameter fresh (default True) is set to False, a previously 
+		cached value will be used if available.
+
+		:param fresh: fetch a new value or use cached response
+		:type fresh: boolean
 		:returns: list of strings
 		"""
-		return self._sendAndReceivePayload('getprofiles').rstrip(';').split(';')
+		if fresh or self._profiles == []:
+			self._profiles = self._sendAndReceivePayload('getprofiles').rstrip(
+				';').split(';')
+		return self._profiles
```

Comment:
There's some changed functionality creeping in with this commit, such as the `fresh` parameters, but your code quality is excellent and you've waited long enough so I'm not going to ask you to break it apart!

Reply:
Thanks! 😃